src/relays/TrustedRelay.sol

```diff
@@ -0,0 +1,141 @@
+    mapping (address => uint256) public signers; // Trusted signers
+    
+    DaiJoinLike            public immutable daiJoin;
+    TokenLike              public immutable dai;
+    WormholeOracleAuthLike public immutable oracleAuth;
+    WormholeJoinLike       public immutable wormholeJoin;
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event SignersAdded(address[] signers);
+    event SignersRemoved(address[] signers);
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "TrustedRelay/non-authed");
+        _;
+    }
+
+    constructor(address _oracleAuth, address _daiJoin) {
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+        oracleAuth = WormholeOracleAuthLike(_oracleAuth);
+        daiJoin = DaiJoinLike(_daiJoin);
+        dai = daiJoin.dai();
+        wormholeJoin = oracleAuth.wormholeJoin();
+    }
+
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function addSigners(address[] calldata signers_) external auth {
+        for(uint i; i < signers_.length; i++) {
+            signers[signers_[i]] = 1;
+        }
+        emit SignersAdded(signers_);
+    }
+
+    function removeSigners(address[] calldata signers_) external auth {
+        for(uint i; i < signers_.length; i++) {
+            signers[signers_[i]] = 0;
+        }
+        emit SignersRemoved(signers_);
+    }
+
+    /**
+     * @notice Gasless relay for the Oracle fast path
+     * The final signature is ABI-encoded `hashGUID`, `maxFeePercentage`, `gasFee`, `expiry`
+     * @param wormholeGUID The wormhole GUID
+     * @param signatures The byte array of concatenated signatures ordered by increasing signer addresses.
+     * Each signature is {bytes32 r}{bytes32 s}{uint8 v}
+     * @param maxFeePercentage Max percentage of the withdrawn amount (in WAD) to be paid as fee (e.g 1% = 0.01 * WAD)
+     * @param gasFee DAI gas fee (in WAD)
+     * @param expiry Maximum time for when the query is valid
+     * @param v Part of ECDSA signature
+     * @param r Part of ECDSA signature
+     * @param s Part of ECDSA signature
+     */
+    function relay(
+        WormholeGUID calldata wormholeGUID,
+        bytes calldata signatures,
+        uint256 maxFeePercentage,
+        uint256 gasFee,
+        uint256 expiry,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external {
+        require(block.timestamp <= expiry, "TrustedRelay/expired");
+        bytes32 signHash = keccak256(abi.encodePacked(
+            "\x19Ethereum Signed Message:\n32", 
+            keccak256(abi.encode(getGUIDHash(wormholeGUID), maxFeePercentage, gasFee, expiry))
+        ));
+        address recovered = ecrecover(signHash, v, r, s);
+        require(signers[recovered] == 1, "TrustedRelay/invalid-signature");
```

Comment:
Should we allow the `receiver` to sign in addition to the trusted relayers?

Reply:
Yes, I don't see why not

Reply:
Great. I think this will help if user gets censored by the trusted network for any reason.